File: inspect_excel.js
const ExcelJS = require('exceljs');

async function inspectFormulas() {
    const workbook = new ExcelJS.Workbook();
    await workbook.xlsx.readFile('DVT260006-PTC_00.xlsm');

    workbook.eachSheet((sheet, id) => {
        if (sheet.name === 'PC' || sheet.name.includes('Custos')) {
            console.log(`\n=== PLANILHA: ${sheet.name} ===`);
            sheet.eachRow((row, rowNumber) => {
                let rowData = `Row ${rowNumber}: `;
                let hasImportantData = false;

                row.eachCell((cell, colNumber) => {
                    if (cell.type === ExcelJS.ValueType.Formula) {
                        hasImportantData = true;
                        rowData += `[Col ${colNumber}: FORMULA -> ${cell.formula} = ${cell.result}] `;
                    } else if (cell.value && typeof cell.value === 'string') {
                        // Print labels that might indicate totals, margins, taxes
                        const valLower = cell.value.toLowerCase();
                        if (valLower.includes('imposto') || valLower.includes('margem') ||
                            valLower.includes('total') || valLower.includes('lucro') ||
                            valLower.includes('custo') || valLower.includes('venda') ||
                            valLower.includes('desconto')) {
                            hasImportantData = true;
                            rowData += `[Col ${colNumber}: LABEL -> ${cell.value}] `;
                        }
                    } else if (cell.value && typeof cell.value === 'number') {
                        rowData += `[Col ${colNumber}: NUM -> ${cell.value}] `;
                    }
                });

                if (hasImportantData) console.log(rowData);
            });
        }
    });
}

inspectFormulas().catch(err => console.error(err));

File: inspect_excel.py
import openpyxl

def inspect_excel():
    print("Opening workbook...")
    wb = openpyxl.load_workbook('DVT260006-PTC_00.xlsm', data_only=False)
    
    for sheet_name in wb.sheetnames:
        if 'PC' in sheet_name or 'Custos' in sheet_name or 'DP' in sheet_name:
            print(f"\n=== SHEET: {sheet_name} ===")
            sheet = wb[sheet_name]
            
            for row in sheet.iter_rows():
                row_data = f"Row {row[0].row}: "
                has_data = False
                for cell in row:
                    if cell.data_type == 'f': # Formula
                        has_data = True
                        row_data += f"[Col {cell.column_letter}: FORMULA -> {cell.value}] "
                    elif isinstance(cell.value, str):
                        val_lower = cell.value.lower()
                        if any(kw in val_lower for kw in ['imposto', 'margem', 'total', 'lucro', 'custo', 'venda', 'desconto', 'bdi', 'encargo']):
                            has_data = True
                            row_data += f"[Col {cell.column_letter}: LABEL -> {cell.value}] "
                    elif isinstance(cell.value, (int, float)):
                        row_data += f"[Col {cell.column_letter}: NUM -> {cell.value}] "
                
                if has_data:
                    print(row_data)

if __name__ == "__main__":
    inspect_excel()
